mailbox/api/src/test/java/org/apache/james/mailbox/model/MessageRangeTest.java

```diff
@@ -23,55 +23,133 @@
-        assertThat(ranges).containsExactly(MessageRange.from(1));
+        assertThat(ranges).containsExactly(MessageRange.from(MessageUid.of(1)));
     }
     
     @Test
     public void splitTenElementsRangeShouldReturn4Ranges() {
-        MessageRange range = MessageRange.range(1,10);
+        MessageRange range = MessageRange.range(MessageUid.of(1),MessageUid.of(10));
         List<MessageRange> ranges = range.split(3);
         assertThat(ranges).containsExactly(
-                MessageRange.range(1, 3), 
-                MessageRange.range(4, 6), 
-                MessageRange.range(7, 9), 
-                MessageRange.one(10));
+                MessageRange.range(MessageUid.of(1), MessageUid.of(3)), 
+                MessageRange.range(MessageUid.of(4), MessageUid.of(6)), 
+                MessageRange.range(MessageUid.of(7), MessageUid.of(9)), 
+                MessageUid.of(10).toRange());
+    }
+
+    @Test
+    public void includeShouldBeTrueWhenAfterFrom() {
+        MessageRange range = MessageRange.from(MessageUid.of(3));
+        boolean actual = range.includes(MessageUid.of(5));
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void includeShouldBeFalseWhenBeforeFrom() {
+        MessageRange range = MessageRange.from(MessageUid.of(3));
+        boolean actual = range.includes(MessageUid.of(1));
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    public void includeShouldBeTrueWhenEqualsFrom() {
+        MessageRange range = MessageRange.from(MessageUid.of(3));
+        boolean actual = range.includes(MessageUid.of(3));
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void includeShouldBeFalseWhenDifferentOne() {
+        MessageRange range = MessageUid.of(3).toRange();
+        boolean actual = range.includes(MessageUid.of(1));
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    public void includeShouldBeTrueWhenEqualsOne() {
+        MessageRange range = MessageUid.of(3).toRange();
+        boolean actual = range.includes(MessageUid.of(3));
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void includeShouldBeFalseWhenBeforeRange() {
+        MessageRange range = MessageRange.range(MessageUid.of(3), MessageUid.of(6));
+        boolean actual = range.includes(MessageUid.of(1));
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    public void includeShouldBeTrueWhenEqualsFromRange() {
+        MessageRange range = MessageRange.range(MessageUid.of(3), MessageUid.of(6));
+        boolean actual = range.includes(MessageUid.of(3));
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void includeShouldBeTrueWhenInRange() {
+        MessageRange range = MessageRange.range(MessageUid.of(3), MessageUid.of(6));
+        boolean actual = range.includes(MessageUid.of(4));
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void includeShouldBeTrueWhenEqualsToRange() {

```

Comment:
I agree. It is. This test will not improve code coverage. But it will better explain specification, and make it obvious....